packages/wrangler/src/parse.ts

```diff
@@ -0,0 +1,178 @@
+import { readFile as readFile_ } from "node:fs/promises";
+import TOML from "@iarna/toml";
+import { formatMessagesSync } from "esbuild";
+import type { PartialMessage, Location } from "esbuild";
+
+/**
+ * Formats a message using esbuild's pretty-print algorithm.
+ */
+export function formatMessage(
+  message: PartialMessage,
+  kind: "warning" | "error" = "error",
+  label?: string
+): string {
+  const lines = formatMessagesSync([message], {
+    color: true,
+    kind: kind,
+    terminalWidth: process.stderr.columns,
+  });
+  if (label) {
+    lines[0] = lines[0].replace(
+      kind.toUpperCase(),
+      `${label.toUpperCase()} ${kind.toUpperCase()}`
+    );
+  }
+  return lines.join("\n");
+}
+
+/**
+ * Formats and prints a message to stderr.
+ */
+export function printMessage(
+  message: PartialMessage,
+  kind?: "warning" | "error",
+  label?: string
+): void {
+  const text = formatMessage(message, kind, label);
+  process.stderr.write(text);
```

Comment:
Why write to `stderr` rather than using `console.error()`?
Also we need to be careful since if the error is inside an Ink component (e.g. `Dev`) then the Ink renderer and this stderr can come into conflict. At the very least I would recommend manually testing this scenario.